index.js

```diff
@@ -174,6 +189,10 @@ Filter.prototype.build = function() {
 
       return result;
     }));
+  }).then(function(completedResult) {
+    return allSettled(pendingWork).then(function() {
+      return Promise.all(completedResult.concat(pendingWork));
```

Comment:
If there is an error, this currently hides it.  To correct this we must throw (likely the first exception).

Most likely something like

```js
allSettled(pending).then(results => {
  result.forEach(result => {
    if (result.state === 'rejected') { throw result.reason; }
  });
})
```

---

Be sure to add a test covering this case.